pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentselector/SegmentSelectorFactory.java

```diff
@@ -17,19 +17,25 @@
  * under the License.
  */
 package org.apache.pinot.broker.routing.segmentselector;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
 import org.apache.pinot.spi.config.table.TableConfig;
 import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.utils.builder.TableNameBuilder;
 
 
 public class SegmentSelectorFactory {
   private SegmentSelectorFactory() {
   }
 
-  public static SegmentSelector getSegmentSelector(TableConfig tableConfig) {
+  public static SegmentSelector getSegmentSelector(TableConfig tableConfig,
+      ZkHelixPropertyStore<ZNRecord> propertyStore) {
+    String tableNameWithType =
+        TableNameBuilder.forType(tableConfig.getTableType()).tableNameWithType(tableConfig.getTableName());
```

Comment:
Table name in table config always have the suffix
```suggestion
    String tableNameWithType = tableConfig.getTableName();
```

Reply:
fixed